src/components/stops/OpenStreetMap.svelte

```diff
@@ -0,0 +1,144 @@
+<script>
+	import { onMount, onDestroy } from 'svelte';
+	import { browser } from '$app/environment';
+
+	/**
+	 * @typedef {Object} Props
+	 * @property {number} lat - Latitude of the stop
+	 * @property {number} lon - Longitude of the stop
+	 * @property {string} name - Name of the stop
+	 * @property {boolean} [showControls=true] - Whether to show map controls
+	 * @property {number} [zoom=15] - Initial zoom level of the map
+	 */
+
+	/** @type {Props} */
+	let { lat, lon, name, showControls = true, zoom = 15 } = $props();
+
+	let mapElement;
+	let map;
+	let marker;
+	let leafletLoaded = false;
+
+	onMount(() => {
+		if (browser) {
+			if (typeof window.L !== 'undefined') {
+				leafletLoaded = true;
+				initMap();
+			} else {
+				loadLeaflet().then(() => {
+					leafletLoaded = true;
+					initMap();
+				});
+			}
+		}
+	});
+
+	onDestroy(() => {
+		if (browser && map) {
+			map.remove();
+		}
+	});
+
+	function loadLeaflet() {
```

Comment:
I don't want to see a wholly separate mapping library get loaded for this feature.